.github/workflows/publish.yml

```diff
@@ -1,28 +1,36 @@
-name: Upload Package to Pypi
+name: Upload Package to PyPI and TestPyPI
 
 on:
-  release:
+  push:
   workflow_dispatch:
 
 jobs:
   deploy:
-
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v2
     - name: Set up Python
       uses: actions/setup-python@v2
       with:
-        python-version: '3.x'
+        python-version: '3.9'
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
         pip install -r test-requirements.txt
-    - name: Build and publish
-      env:
-        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
-        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
-      run: |
-        python -m build
-        twine upload dist/*
+    - name: Build package
+      run:  python -m build
+    - name: Publish to Test PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: __token__
+        password: ${{ secrets.TEST_PYPI_TOKEN }}
+        repository_url: https://test.pypi.org/legacy/
+    - name: Publish to PyPI
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        password: ${{ secrets.TEST_PYPI_TOKEN }}
```

Comment:
Does this need to be a different token, or is the same used for test and release?

Reply:
These are different tokens (and I already created them and added them to github)

Reply:
Let me rephrase. Should we be using TEST_PYPI_TOKEN here or a non-test token?

Reply:
Ah! the snippet here doesn't show enough of the code, so I didn't realize what you meant.  Good catch, this line should be using the non-test one, will update